### DISCUSSION TITLE
Midi controller functionality

### DISCUSSION THREAD

**Discussion by wujke** [rerdavies/pipedal#75] [Q&A] [answered]
Hi, I'm making midi controller with arduino. It's in prototype phase on breadboard and it looks it's working as it should. Here is the question. I can't find a way to change bank/preset with midi signal. Is there a way to do this?
Thanks

**Comment by rerdavies** [comment]
MIDI program changes aren't implemented yet. Give me a week or two, and
I'll add it. (Busy time of year).

**Comment by rerdavies** [comment, marked as answer]
Added in v1.0.18:

- Program changes via MIDI Program and Bank Select messages.
- next/previous program via MIDI note or control triggers. Use the System MIDI Bindings item in the settings dialog to configure which notes or controls trigger selection of the next or previous program.

**Reply by maarthome** [reply]
That's great :-). Thanks!
